otter/convergence/service.py

```diff
@@ -114,6 +115,10 @@ def execute_convergence(tenant_id, group_id, log,
     launch_config = manifest['launchConfiguration']
     now = yield Effect(Func(time.time))
 
+    group_status = ScalingGroupStatus.lookupByName(manifest['status'])
+    if group_status == ScalingGroupStatus.DELETING:
+        group_state.desired = 0
```

Comment:
> Doesn't the controller code already set the desired to 0 when deleting the group?

No. The controller only sets DELETING status. This code does not persist the updated desired state. It only uses it when calculating the desired group state. I agree it leaks side-effect. I could instead do 
```python
group_status = ScalingGroupStatus.lookupByName(manifest['status'])
desired_group_state = get_desired_group_state(
   group_id, launch_config, 
   0 if group_status == ScalingGroupStatus.DELETING else group_state.desired)
```
This doesn't look like having side-effect and I think it's clearer. What do you think?